src/microbe_stage/editor/MicrobeEditorGUI.cs

```diff
@@ -978,6 +977,50 @@ public void UpdatePatchDetails(Patch patch)
         UpdateReportTabPatchName(TranslationServer.Translate(patch.Name));
     }
 
+    /// <summary>
+    ///   Updates the existing part selection buttons with the correct properties based on their Node names.
+    /// </summary>
+    public void UpdateMicrobePartSelections()
+    {
+        foreach (var organelle in SimulationParameters.Instance.GetAllOrganelles())
+        {
+            foreach (var partSelection in placeablePartSelectionElements)
+            {
+                if (partSelection.Name == organelle.InternalName)
+                {
+                    partSelection.PartName = organelle.Name;
+                    partSelection.MPCost = organelle.MPCost;
+
+                    partSelection.PartIcon = !string.IsNullOrEmpty(organelle.IconPath) ?
+                        GD.Load<Texture>(organelle.IconPath) :
+                        GD.Load<Texture>("res://assets/textures/gui/bevel/TestIcon.png");
+
+                    partSelection.Connect(
```

Comment:
What if this connection is already done? I think there should be 2 steps that are done at separate times: one is to do the general setup namely this connection and the icon loading above. And the second part would be to update the name and cost as those might change.

Reply:
I think `IsConnected` could be used to check if the signal already connected, and for the icon instead of loading it here it could be preloaded in `OrganelleDefinition` so it doesn't have to be loaded at runtime. That way I think it should make them safe to be in one method.